albumentations/augmentations/geometric/transforms.py

```diff
@@ -2038,3 +2039,146 @@ def get_params(self) -> dict[str, D4Type]:
+        always_apply (bool | None): If true, the transform will always be applied.
+
+    Targets:
+        image, mask
+
+    Image types:
+        uint8
+    """
+
+    def __init__(
+        self,
+        n_grid_width: int,
+        n_grid_height: int,
+        magnitude: int,
+        p: float = 1.0,
+        always_apply: bool | None = None,
+    ):
+        super().__init__(p=p, always_apply=always_apply)
+        self.n_grid_width = n_grid_width
+        self.n_grid_height = n_grid_height
+        self.magnitude = abs(magnitude)
+
+    def calculate_dimensions(
+        self,
+        width_of_square,
+        height_of_square,
+        width_of_last_square,
+        height_of_last_square,
+    ):
+        dimensions = []
+        for vertical_tile in range(self.n_grid_width):
+            for horizontal_tile in range(self.n_grid_height):
+                x1 = horizontal_tile * width_of_square
+                y1 = vertical_tile * height_of_square
+                x2 = x1 + (
+                    width_of_last_square
+                    if horizontal_tile == self.n_grid_height - 1
+                    else width_of_square
+                )
+                y2 = y1 + (
+                    height_of_last_square
+                    if vertical_tile == self.n_grid_width - 1
+                    else height_of_square
+                )
+                dimensions.append([x1, y1, x2, y2])
+
+        return dimensions
+
+    def calculate_polygons(self, dimensions, horizontal_tiles, vertical_tiles):
+        polygons = []
+        for x1, y1, x2, y2 in dimensions:
+            polygons.append([x1, y1, x1, y2, x2, y2, x2, y1])
+
+        last_column = [
+            (horizontal_tiles - 1) + horizontal_tiles * i for i in range(vertical_tiles)
+        ]
+        last_row = range(
+            (horizontal_tiles * vertical_tiles) - horizontal_tiles,
+            horizontal_tiles * vertical_tiles,
+        )
+
+        polygon_indices = []
+        for i in range((vertical_tiles * horizontal_tiles) - 1):
+            if i not in last_row and i not in last_column:
+                polygon_indices.append(
+                    [i, i + 1, i + horizontal_tiles, i + 1 + horizontal_tiles]
+                )
+
+        for a, b, c, d in polygon_indices:
+            dx = random.randint(-self.magnitude, self.magnitude)
+            dy = random.randint(-self.magnitude, self.magnitude)
+
+            polygons[a][4] += dx
+            polygons[a][5] += dy
+            polygons[b][2] += dx
+            polygons[b][3] += dy
+            polygons[c][6] += dx
+            polygons[c][7] += dy
+            polygons[d][0] += dx
+            polygons[d][1] += dy
+
+        return polygons
+
+    def generate_mesh(self, polygons, dimensions):
+        return [[dimensions[i], polygons[i]] for i in range(len(dimensions))]
+
+    def distort_image(self, image: np.ndarray, generated_mesh: List[List]):
+        image = Image.fromarray(image)
```

Comment:
We avoid brigning heavy dependencies to the library when existing could work for the task.

=>

`Pillow` => `opencv`